zipline/pipeline/loaders/blaze/core.py

```diff
@@ -729,25 +729,36 @@ def __new__(cls,
             expr,
             deltas,
             checkpoints,
-            odo_kwargs or {},
+            frozenset((odo_kwargs or {}).items()),
         )
 
     def __repr__(self):
         # If the expressions have _resources() then the repr will
         # drive computation so we take the str here.
-        cls = type(self)
-        return super(ExprData, cls).__repr__(cls(
+        return repr(_expr_data_base(
             str(self.expr),
             str(self.deltas),
             str(self.checkpoints),
-            self.odo_kwargs,
+            dict(self.odo_kwargs),
         ))
 
-    def __hash__(self):
-        return id(self)
+    @staticmethod
+    def _expr_eq(a, b):
+        return a is b is None or a.isidentical(b)
 
     def __eq__(self, other):
-        return self is other
+        if not isinstance(other, ExprData):
+            return NotImplemented
+
+        return (
+            self.expr.isidentical(other.expr) and
+            self._expr_eq(self.deltas, other.deltas) and
+            self._expr_eq(self.checkpoints, other.checkpoints) and
+            self.odo_kwargs == other.odo_kwargs
+        )
+
+    def __ne__(self, other):
```

Comment:
I think this is the default behavior. Is there a reason to implement this explicitly?

```
In [16]: class Foo(object):
    ...:
    ...:     def __eq__(self, other):
    ...:         print("calling eq")
    ...:         return True
    ...:
    ...:

In [17]: Foo() == Foo()
calling eq
Out[17]: True

In [18]: Foo() != Foo()
calling eq
Out[18]: False
```

Reply:
I believe `tuple` changes this:
```python
In [9]: class T(tuple):
   ...:     def __eq__(self, other):
   ...:         print('eq')
   ...:         return super(T, self).__eq__(other)
   ...:     
   ...:     

In [10]: t = T((1, 2, 3))

In [11]: t == t
eq
Out[11]: True

In [12]: t != t
Out[12]: False

```

Reply:
Ah. Can we add a comment here noting that we need a `__ne__` to override the one we get from being a tuple subclass?